SilMock/tests/Google/Service/Gmail/Resource/UsersSettingsDelegatesTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use PHPUnit\Framework\Assert;
+use SilMock\Google\Service\Gmail\Resource\UsersSettingsDelegates;
+use SilMock\Google\Service\GoogleFixtures;
+
+class UsersSettingsDelegatesTest extends PHPUnit\Framework\TestCase
+{
+    public $dataFile = DATAFILE4;
+    
+    protected function setUp(): void
+    {
+        $this->emptyFixturesDataFile();
+    }
+    
+    private function emptyFixturesDataFile()
+    {
+        $fixturesClass = new GoogleFixtures($this->dataFile);
+        $fixturesClass->removeAllFixtures();
+    }
+    
+    protected function tearDown(): void
+    {
+        $this->emptyFixturesDataFile();
+    }
+    
+    public function testListUsersSettingsDelegates()
+    {
+        // Arrange:
+        $accountEmail = 'john_smith@example.org';
+        $delegateEmail = 'mike_manager@example.org';
+        
+        // Act
+        $before = $this->getDelegatesForAccount($accountEmail);
+        $this->delegateAccessToAccountBy($accountEmail, $delegateEmail);
+        $after = $this->getDelegatesForAccount($accountEmail);
+        
+        // Assert:
+        $delegatesBefore = $before->getDelegates();
+        Assert::assertEmpty($delegatesBefore);
+        
+        $delegatesAfter = $after->getDelegates();
+        Assert::assertNotEmpty($delegatesAfter);
+        
+        $foundExpectedDelegate = false;
+        foreach ($delegatesAfter as $delegate) {
```

Comment:
Why foreach this? It's based on it's own data file, there is only one delegate. Assert the size, assert they are equal. Why complicate this?

Reply:
Google doesn't pass back 0 indexed array... it's 1. Who knows if interface will change.